plotutil/xyer.go

```diff
@@ -0,0 +1,71 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plotutil
+
+import (
+	"fmt"
+
+	"github.com/gonum/plot/plotter"
+)
+
+// VecXY implements plotter.XYer.
+type VecXY struct {
+	X, Y []float64
+}
+
+// ZipXY is a convenience function to build a VecXY from slices with length checks.
+// Panics on length mismatch.
+func ZipXY(x, y []float64) plotter.XYer {
+	if len(y) != len(x) {
+		panic(fmt.Sprintf("VecXY length mismatch %d != %d", len(x), len(y)))
+	}
+	return &VecXY{X: x, Y: y}
+}
+
+// Len returns the length of the VecXY object.
```

Comment:
Please use complete sentences: "Len returns the length of the T value." (I would like T to be `XYSlices`)